src/Transport/RestTransport.php

```diff
@@ -109,24 +111,34 @@ public static function build(string $apiEndpoint, string $restConfigPath, array
      */
     public function startUnaryCall(Call $call, array $options)
     {
+        $requestLog = null;
         $headers = self::buildCommonHeaders($options);
+        $request =  $this->requestBuilder->build(
+            $call->getMethod(),
+            $call->getMessage(),
+            $headers
+        );
+
+        if ($this->logger) {
+            $requestLog = $this->logRequest($request);
```

Comment:
I am using this to have access to the `timestamp` and calculate the latency. But this is something that we can move to the `LoggerMiddleware` using a `then` inside that middleware.